scripts/ci/run-jenkins.sh

```diff
@@ -6,8 +6,14 @@ export TEST_HARNESS_VERBOSE=1
 
 if [[ ${label} == 'osx-i386' ]]; then EXTRA_CONF_FLAGS="--with-libgdiplus=/Library/Frameworks/Mono.framework/Versions/Current/lib/libgdiplus.dylib --enable-nls=no --build=i386-apple-darwin11.2.0"; fi
 if [[ ${label} == 'osx-amd64' ]]; then EXTRA_CONF_FLAGS="--with-libgdiplus=/Library/Frameworks/Mono.framework/Versions/Current/lib/libgdiplus.dylib --enable-nls=no"; fi
-if [[ ${label} == 'w32' ]]; then PLATFORM=Win32; EXTRA_CONF_FLAGS="--host=i686-pc-mingw32"; export MONO_EXECUTABLE="`cygpath -u ${WORKSPACE}\\\msvc\\\Win32\\\bin\\\Release_SGen\\\mono-sgen.exe`";fi
-if [[ ${label} == 'w64' ]]; then PLATFORM=x64; EXTRA_CONF_FLAGS="--host=i686-pc-mingw32"; export MONO_EXECUTABLE="`cygpath -u ${WORKSPACE}\\\msvc\\\x64\\\bin\\\Release_SGen\\\mono-sgen.exe`"; fi
+if [[ ${label} == 'w32' ]]; then PLATFORM=Win32; EXTRA_CONF_FLAGS="--host=i686-pc-mingw32"; fi
+if [[ ${label} == 'w64' ]]; then PLATFORM=x64; EXTRA_CONF_FLAGS="--host=i686-pc-mingw32"; fi
+
+if [[ ${label} == w* ]]; then
+	export MONO_EXECUTABLE="`cygpath -u ${WORKSPACE}\\\msvc\\\\${PLATFORM}\\\bin\\\Release_SGen\\\mono-sgen.exe`"
+	# without this dir in PATH the P/Invoke tests in mono/tests/ won't find the MSVC built libtest.dll
+	export PATH="$PATH:`cygpath -u ${WORKSPACE}\\\msvc\\\\${PLATFORM}\\\bin\\\Release`"
```

Comment:
I think a cleaner way would be to add a [dllmap](http://www.mono-project.com/docs/advanced/pinvoke/dllmap/) entry to [tests-config.in](https://github.com/mono/mono/blob/master/mono/tests/tests-config.in) for libtest.dll on Windows that points to the MSVC built library.

I'm not 100% sure whether this works or breaks when running against the cygwin mono where the library is in the same folder (i.e. does it first look in the dllmap location and fail if not found, or local folder first?), but that should be easy to try.